luisy/targets.py

```diff
@@ -137,6 +159,166 @@ def _try_to_upload(self, overwrite=False):
         )
 
 
+class CloudTarget(LuisyTarget):
+
+    def __init__(self, path, **kwargs):
+        LuisyTarget.__init__(
+            self,
+            path=path,
+            format=None,
+            is_tmp=False
+        )
+        self.kwargs = kwargs
+
+
+class SparkTarget(CloudTarget):
+    """
+    this abstract class is for targets working with spark instances.
+    """
+    @property
+    def spark(self):
+        try:
+            return Config().spark
+        except AttributeError:
+            raise AttributeError(
+                "spark session was not found in config. Make sure to have all the databricks "
+                "parameters set in order to start the spark session!"
+            )
+
+
+class DeltaTableTarget(SparkTarget):
+    # TODO: Can we get rid of fileending
+    file_ending = 'DeltaTable'
```

Comment:
why exaclty do we need this fileending ?

Reply:
I think it is okay that we have that "fake" fileending for the delta table here, for now. But actually, in  `luisy.hashes.HashMapping.clean_up`  we could check whether the table exists in the cloud, and if not delete its hash from the local hashes file. But I see that this would mess up the original `--upload` and `--download` strategy in luisy. It is all getting quite complex, so maybe lets use and the new functionality a while before extending it.